crates/hyperqueue/src/server/event/journal/stream.rs

```diff
@@ -83,11 +93,28 @@ async fn streaming_process(
                            And while this read is performed, we cannot allow modification of the file,
                            so the writing to the file has to be paused anyway */
                         writer.flush()?;
-                        let mut reader = JournalReader::open(log_path)?;
+                        let mut reader = JournalReader::open(journal_path)?;
                         for event in &mut reader {
                             tx.send(event?).unwrap()
                         }
-                    }
+                    },
+                    Some(EventStreamMessage::PruneJournal { live_jobs, live_workers, callback })  => {
+                        writer.flush()?;
+                        let mut tmp_path: OsString = journal_path.into();
+                        tmp_path.push(".tmp");
+                        let tmp_path: PathBuf = tmp_path.into();
+                        {
+                            let mut reader = JournalReader::open(journal_path)?;
+                            let mut writer = JournalWriter::create_or_append(&tmp_path, None)?;
+                            if let Err(e) = prune_journal(&mut reader, &mut writer, &live_jobs, &live_workers) {
+                                remove_file(&tmp_path)?;
+                                return Err(e.into())
+                            }
+                        }
+                        rename(&tmp_path, &journal_path)?;
+                        writer = JournalWriter::create_or_append(journal_path, None)?;
```

Comment:
Shouldn't we drop the writer before doing the rename? Otherwise the `writer` will be closed while the file it deals with was already overwritten. It's probably fine, but one never knows with file descriptors...